packages/asc-ui/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { ChevronLeft, ChevronRight } from '@datapunt/asc-assets'
+import PaginationStyle, {
+  Props,
+  PageNumber,
+  PreviousButton,
+  NextButton,
+} from './PaginationStyle'
+
+const Pagination: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
+  page = 1,
+  collectionSize,
+  pageSize = 10,
+  onPageChange,
+  ...otherProps
+}) => {
+  const [currentPage, setCurrentPage] = React.useState<number>(page)
+  const totalPages = Math.ceil(collectionSize / pageSize)
+  const pageCluster = [currentPage]
+  if (currentPage > 2) {
+    pageCluster.unshift(currentPage - 1)
+  }
+  if (totalPages <= 5 || currentPage < totalPages - 1) {
+    pageCluster.push(currentPage + 1)
+  }
+
+  React.useEffect(() => {
+    if (onPageChange !== undefined) onPageChange(currentPage)
+  }, [currentPage, onPageChange])
+
+  const handlePrevious = () => {
```

Comment:
small thing, but could we rename this to `onHandlePrevious`? That would make it a bit more clear that this is an event handler

Reply:
I settled on `onPrevious`.